### PR TITLE
feat: add loguru for logging

### PR DESCRIPTION
**Pull request by s0nicboOm**
Support logging with Loguru library.
Link: https://github.com/Delgan/loguru

### CODE REVIEW

File: numaprom/__init__.py [hunk 1]
```diff
@@ -1,36 +1,54 @@
...
+        filter_levels["numaprom"] = "INFO"
+        logger.add(sink=sink, level="INFO", colorize=True, filter=filter_levels)
+
+    logger.info("Starting Logger...")
     return logger
 
 
-__all__ = ["UnifiedConf", "MetricConf", "AppConf", "DataConf", "get_logger"]
+# get Logger
+_LOGGER = __get_logger()
```
Comment (ab93) [inline numaprom/__init__.py:52]: let's rename it to `LOGGER` since it is being imported everywhere.

File: numaprom/__init__.py [hunk 2]
```diff
@@ -1,36 +1,54 @@
...
+    logger.remove()
+
+    # define log sink
+    sink = sys.stderr
+
+    # define library log levels
+    filter_levels = {
+        "watchdog": "ERROR",
+        "pytorch_lightning": "ERROR",
+        "pynumaflow": "ERROR",
```
Comment (ab93) [inline numaprom/__init__.py:36]: let's bring this down to warning

**Review by ab93 — COMMENTED** [review]
Where are we setting the log format?

File: numaprom/__init__.py (hunk 1 above)
Comment (s0nicboOm) [reply numaprom/__init__.py]: FIxed

File: numaprom/__init__.py (hunk 2 above)
Comment (s0nicboOm) [reply numaprom/__init__.py]: Set it to WARNING now.

**Review by ab93 — CHANGES_REQUESTED** [review]
Can you fix the unit tests @s0nicboOm ?

**Comment by s0nicboOm** [comment]
> Where are we setting the log format?
formatter: `formatter = "<green>{time:YYYY-MM-DD HH:mm:ss.SSS}</green> | <level>{level: <8}</level> | `<level>{message}</level>"`
image

**Comment by s0nicboOm** [comment]
> > Where are we setting the log format?
> > formatter: `formatter = "<green>{time:YYYY-MM-DD HH:mm:ss.SSS}</green> | <level>{level: <8}</level> | `{message}"`
> > image

Different logs also have different colors: 
image

**Review by ab93 — APPROVED** [review]

**Review by nkoppisetty — APPROVED** [review]

**Pull request merged**